### ISSUE TITLE
Shadow casting/receiving API

### ISSUE THREAD

**Issue by kvark** [kvark/three-rs#38]
Three.js has a pair of simple boolean flags per object:
```
object3d.castShadow = true;
object3d.receiveShadow = false;
```
This doesn't exactly work with remote nodes that we have, and is not flexible enough.
One possible scheme:
```rust
light.clear_casters()l
light.add_caster(&mesh1);
light.add_caster(&mesh2);
```
If the light affects an object, then the object automatically becomes the receiver. I don't think it ever makes sense to be lit but not receive the shadow.

**Comment by vitvakatu** [comment]
I like the current proposal with `add_caster`, it seems flexible enough. @alteous?

**Comment by alteous** [comment]
I can't offer an informed opinion. Let me play around with the shadow system for a little while and I'll get back to you.

Looking at the aviator demo, it seems that only the aeroplane casts a shadow which seems odd to me.

**Comment by expenses** [comment]
Do shadows work at all (without getting into lower level things like shaders) right now?

**Comment by kvark** [comment]
@expenses the `lights` example shows how the lower level thing works. You cycle through the light types with left/right, and you can see both the shadow casted, and the shadow map itself rendered.

**Comment by expenses** [comment]
Right, yeah, sorry. When i was having trouble I was using a PBR shader, which doesn't seem to cast shadows.
